tools/filetools.go

```diff
@@ -121,3 +126,250 @@ func VerifyFileHash(oid, path string) error {
 
 	return nil
 }
+
+// FilenamePassesIncludeExcludeFilter returns whether a given filename passes the include / exclude path filters
+// Only paths that are in includePaths and outside excludePaths are passed
+// If includePaths is empty that filter always passes and the same with excludePaths
+// Both path lists support wildcard matches
+func FilenamePassesIncludeExcludeFilter(filename string, includePaths, excludePaths []string) bool {
+	if len(includePaths) == 0 && len(excludePaths) == 0 {
+		return true
+	}
+
+	if len(includePaths) > 0 {
+		matched := false
+		for _, inc := range includePaths {
+			matched = FileMatch(inc, filename)
+			if matched {
+				break
+			}
+		}
+		if !matched {
```

Comment:
I don't understand your logic, sorry. You'll need to write me the complete block as you see it because to me that doesn't seem to equate to the same result.